kiali/kiali
# Autodetect sidecar injector configmap

### Describe the change

Autodetects the istio sidecar injector configmap. Moves `network` from the cluster object to the controlplane since this is where the info comes from.

### Steps to test the PR

1. Deploy kiali and istio
2. Set `external_services.istio.istio_sidecar_injector_config_map_name = ""`
3. Ensure kiali still works and there's no errors logs in the kiali server about the sidecar injector configmap
4. Ensure you can see the `network` in the mesh target panel for the istiod:
![Screenshot from 2024-07-01 13-50-34](https://github.com/kiali/kiali/assets/6226732/d0f453b9-2144-4f23-a15d-793111516402)

### Automation testing

Included unit tests

### Issue reference

Fixes #7177
